### PR TITLE
feat: add log writer to the kermit config [AR-1450]

### PR DESCRIPTION
**Pull request by MohammedMokresh**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

without having the log writer in kermit we don't have a way to intercept the logs that been logged in kermit 

### Solutions

add the support to add a custom log writer to kermit to support writing to file 

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logger/build.gradle.kts [hunk 1]
```diff
@@ -43,7 +43,7 @@ kotlin {
     sourceSets {
         val commonMain by getting {
             dependencies {
-                implementation("co.touchlab:kermit:1.0.0")
+                api("co.touchlab:kermit:1.1.1")
```
Comment (alexandreferris) [inline logger/build.gradle.kts:46]: Why the change from `implementation` to `api`?

**Review by alexandreferris — COMMENTED** [review]
Great work! I just have one small question tho 🤔

File: logger/build.gradle.kts (hunk 1 above)
Comment (yamilmedina) [reply logger/build.gradle.kts:46]: I think it's related to having this lib automatically imported on the consumers projects. 

I'm wondering if this could cause a dependency conflict in the future, in case another lib would import this `kermit` lib transitively 🤔

**Review by yamilmedina — APPROVED** [review]
Looks good to me, just a small question that jumped to my head on the possibility to have deps conflicts in the future, but I think you could decide the best approach 🙌  right now I don't see an issue leaving it as api =)

File: logger/build.gradle.kts (hunk 1 above)
Comment (MohammedMokresh) [reply logger/build.gradle.kts:46]: because we are using it in the reloaded as we need to build the log writer from the library , and in theory the other clients will do the same to create a custom log writer

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/EnableLoggingUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.wire.kalium.logic.feature.user
+
+import com.wire.kalium.logic.configuration.UserConfigRepository
+
+
+class EnableLoggingUseCase(
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/EnableLoggingUseCase.kt:6]: Could we extract an interface out of this UseCase ? It will be easy I think in the future for espresso testing in case we want to create Fakes.

You can see an example on `UploadUserAvatarUseCase`

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsLoggingEnabledUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.wire.kalium.logic.feature.user
+
+import com.wire.kalium.logic.StorageFailure
+import com.wire.kalium.logic.configuration.UserConfigRepository
+import com.wire.kalium.logic.kaliumLogger
+
+class IsLoggingEnabledUseCase(
```
Comment (yamilmedina) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/IsLoggingEnabledUseCase.kt:7]: Could we extract an interface out of this UseCase ? It will be easy I think in the future for espresso testing in case we want to create Fakes.

You can see an example on `UploadUserAvatarUseCase`

**Review by yamilmedina — COMMENTED** [review]
Great work, Big Mo! 🪄  Just left some comments about having the Use Cases behind an interface, so We could mock/fake them later for espresso testing 🚀

**Review by yamilmedina — APPROVED** [review]
Good work! thanks for applying the changes 🪄 🚀
